test/internal/laws/Setoid.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var forall = require('./forall');
+
+
+module.exports = forall({
+
+  //  a `equals` a = true
+  reflexivity: function(a) {
+    return a.equals(a);
+  },
+
+  //  a `equals` b = b `equals` a
+  symmetry: function(a, b) {
+    return a.equals(b) === b.equals(a);
+  },
+
+  //  a `equals` b & b `equals` c => a `equals` c
+  transitivity: function(a, b, c) {
+    return !a.equals(b) || !b.equals(c) || a.equals(c);
```

Comment:
Sure. Let me know which of these you like best and we'll go with it:

  - `!a.equals(b) || !b.equals(c) || a.equals(c)`
  - `!(a.equals(b) && b.equals(c)) || a.equals(c)`
  - `a.equals(b) && b.equals(c) ? a.equals(c) : true`